filebeat/input/filestream/input_test.go

```diff
@@ -101,41 +147,41 @@ func createFilestreamTestRunner(b testing.TB, testID string, cfg string, eventLi
 	input, err := p.Manager.Create(c)
 	require.NoError(b, err)
 
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
 	context := v2.Context{
 		Logger:      logger,
 		ID:          testID,
 		Cancelation: ctx,
 	}
 
-	events := make([]beat.Event, 0, eventLimit)
-	connector, eventsDone := newTestPipeline(eventLimit, &events)
-	done := make(chan struct{})
+	connector, events := newTestPipeline(eventLimit, collectEvents)
+	var out []beat.Event
+	if collectEvents {
+		out = make([]beat.Event, 0, eventLimit)
+	}
+	go func() {
+		for event := range events {
+			out = append(out, event)
+		}
+		cancel()
+	}()
```

Comment:
If I understood correctly, when `collectEvents` is set to `false`, nothing will be written to `events` channel, if that is true, then this goroutine does not need to be started.

```suggestion
	connector, events := newTestPipeline(eventLimit, collectEvents)
	var out []beat.Event
	if collectEvents {
		out = make([]beat.Event, 0, eventLimit)
		go func() {
			for event := range events {
				out = append(out, event)
			}
			cancel()
		}()
	}
```

Reply:
@belimawr it needs to be started since we keep using the `events` channel as `done` basically, we still need to know when filestream processed the expected amount of events, we just don't collect them.

Reply:
I see it now. Could you add a comment explaining it? Just looking at the code this is not so obvious.

Reply:
@belimawr I added the comment in https://github.com/elastic/beats/pull/37345/commits/627f900328d3311c1869927771a221cb043798fe